### ISSUE TITLE
[java] DataflowAnomalyAnalysis: DD false positive for arrays

### ISSUE THREAD

**Issue by cobratbq** [pmd/pmd#1251]
**Affects PMD Version:** pmd 6.5.0 executed through maven-pmd-plugin 3.8

**Rule:** [DataflowAnomalyAnalysis](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#dataflowanomalyanalysis) (DD false positive)

**Description:**

The variable 'msg2', which is declared as an array of BigInteger, is treated like a plain variable being assigned multiple times, while in fact assignments are to specific indexes of the array.

**Code Sample demonstrating the issue:**

```java
public class Test {
    public void test() {
        final BigInteger[] msg2 = new BigInteger[11];
        msg2[0] = G1.modPow(x2, OtrCryptoEngine.MODULUS);
        BigInteger[] res = proofKnowLog(x2, 3);
        msg2[1] = res[0];
        msg2[2] = res[1];
        // etc.
    }
}
```

Result:

```
n/a:3:	Found 'DD'-anomaly for variable 'msg2' (lines '3'-'4').
n/a:4:	Found 'DD'-anomaly for variable 'msg2' (lines '4'-'6').
n/a:6:	Found 'DD'-anomaly for variable 'msg2' (lines '6'-'7').
n/a:7:	Found 'DU'-anomaly for variable 'msg2' (lines '7'-'9').
```

**Running PMD through:** *Maven*

**Comment by adangel** [comment]
The rule [DataflowAnomalyAnalysis](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#dataflowanomalyanalysis) has been deprecated in favour of [UnusedAssignment](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedassignment), which is available with PMD 6.26.0.

Therefore this issue wont be fixed.

See also #2647.
